contracts/cw-govmod-single/src/contract.rs

```diff
@@ -0,0 +1,411 @@
+#[cfg(not(feature = "library"))]
+use cosmwasm_std::entry_point;
+use cosmwasm_std::{
+    to_binary, Addr, Binary, CosmosMsg, Deps, DepsMut, Empty, Env, MessageInfo, Response,
+    StdResult, WasmMsg,
+};
+use cw2::set_contract_version;
+use cw_storage_plus::Bound;
+use cw_utils::{Duration, Expiration};
+
+use crate::{
+    error::ContractError,
+    msg::{ExecuteMsg, InstantiateMsg, QueryMsg},
+    proposal::{advance_proposal_id, Proposal, Status, Vote, Votes},
+    query::{ProposalResponse, VoteInfo, VoteListResponse, VoteResponse},
+    state::{Ballot, Config, BALLOTS, CONFIG, PROPOSALS, PROPOSAL_COUNT},
+    threshold::Threshold,
+    utils::{get_total_power, get_voting_power},
+};
+
+const CONTRACT_NAME: &str = "crates.io:cw-govmod-single";
+const CONTRACT_VERSION: &str = env!("CARGO_PKG_VERSION");
+
+/// Default limit for proposal pagination.
+const DEFAULT_LIMIT: u64 = 30;
+
+#[cfg_attr(not(feature = "library"), entry_point)]
+pub fn instantiate(
+    deps: DepsMut,
+    _env: Env,
+    info: MessageInfo,
+    msg: InstantiateMsg,
+) -> Result<Response, ContractError> {
+    set_contract_version(deps.storage, CONTRACT_NAME, CONTRACT_VERSION)?;
+
+    msg.threshold.validate()?;
+
+    let dao = info.sender;
+
+    let config = Config {
+        threshold: msg.threshold,
+        max_voting_period: msg.max_voting_period,
+        only_members_execute: msg.only_members_execute,
```

Comment:
just curious, are there situations in which it's important for non-members to be able to execute?

Reply:
Yes, if you have cron based execution for example (an outside bot that looks for passed proposals and automatically executes them).